### DISCUSSION TITLE
Backup failed because There are no files to be backed up

### DISCUSSION THREAD

**Discussion by ultrono** [spatie/laravel-backup#1554] [Ideas]
Hi,

I've been using this package for ages without issue. I have a Laravel application running version `6.16.5` of this package.

I'm only backing up the database and have a scheduled command defined as followed:

```php
$schedule
    ->command(\Spatie\Backup\Commands\BackupCommand::class, ['--only-db'])
    ->everyFourHours();
```

Again, this has worked without issue for a long time.

Running `php artisan backup:run --only-db` outputs: "Backup failed because There are no files to be backed up.."

Running `php artisan backup:run` works, both files and the database are backed up.

I did see a random issue that recommended I change the config setting `backup.source.files.include` to an empty array. However, that gave the same error as above.

Many Thanks

Rob

**Comment by ultrono** [comment]
Tried upgrading to to `7.8.0`. 

Running `php artisan backup:run --only-db` still gives the same output:

```
[user@domain public_html]$ php artisan backup:run --only-db
Starting backup...
Dumping database tasks...
Determining files to backup...
Backup failed because There are no files to be backed up..
#0 /home/user/public_html/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(163): Spatie\Backup\Exceptions\InvalidBackupJob::noFilesToBeBackedUp()
```

**Comment by PaolaRuby** [comment]
Feel free to make a PR to fix that

**Reply by ultrono** [reply]
That's not overly helpful to be honest. if I was aware of the exact problem I would have fixed it myself or created a PR : At the moment backing up the site files and database works fine. Oddly the database only option causes the described error. There's definitely something amiss that has changed fairly recently. It may well be an issue my end :/ However, the fact @angeljqv appears to be seeing the same issue says there must be something to the issue 🤔

**Comment by coziboy** [comment]
> php artisan backup:run --only-db

same here..
i hope someone will make a PR to fix this issue

**Comment by angeljqv** [comment]
> i hope someone will make a PR to fix this issue

@coziboy mobody gonna make PR for you, make it yourself, don't be lazy and contribute to the community

**Comment by ultrono** [comment]
FYI another person is reporting the same issue at https://github.com/spatie/laravel-backup/issues/1555

**Reply by cesarreyes3** [reply]
Fell free to make a PR fixing that

**Reply by amit-viacon** [reply]
@cesarreyes3 let's make our own package to backup.

It is easier for those who maintain this package to understand and debug the problem. I have tried to find the issue but couldn't find it.
If you can't help, shut your mouth.

**Reply by freekmurze** [reply]
@amit-viacon Thanks for your opinion, but please keep the conversation tone civil.

**Reply by cesarreyes3** [reply]
@amit-viacon of course, you are helping a lot with your comment

And this is a public package, everyone can send a fix, that you don't have the capacity or the intelligence is something else

**Comment by ultrono** [comment]
For anyone looking at this, the issue went away when I upgraded my project to Laravel 9 and used v8 of this package.
